test/parallel/test-runner-run-coverage.mjs

```diff
@@ -0,0 +1,182 @@
+            code: 'ERR_INVALID_ARG_TYPE'
+          });
+        });
+
+      assert.throws(() => run({ coverage: true, branchCoverage: -1 }), { code: 'ERR_OUT_OF_RANGE' });
+      assert.throws(() => run({ coverage: true, branchCoverage: 101 }), { code: 'ERR_OUT_OF_RANGE' });
+
+      run({ files: [], signal: AbortSignal.abort(), coverage: true, branchCoverage: 0 });
+    });
+
+    await it('should only allow an int within range in options.functionCoverage', async () => {
+      [Symbol(), {}, () => {}, [], 0n, 1n, Promise.resolve([]), true, false]
+        .forEach((functionCoverage) => {
+          assert.throws(() => run({ coverage: true, functionCoverage }), {
+            code: 'ERR_INVALID_ARG_TYPE'
+          });
+          assert.throws(() => run({ coverage: true, functionCoverage: [functionCoverage] }), {
+            code: 'ERR_INVALID_ARG_TYPE'
+          });
+        });
+
+      assert.throws(() => run({ coverage: true, functionCoverage: -1 }), { code: 'ERR_OUT_OF_RANGE' });
+      assert.throws(() => run({ coverage: true, functionCoverage: 101 }), { code: 'ERR_OUT_OF_RANGE' });
+
+      run({ files: [], signal: AbortSignal.abort(), coverage: true, functionCoverage: 0 });
+    });
+  });
+
+  await describe('run with coverage', skipIfNoInspector, async () => {
+    await it('should run with coverage', async () => {
+      const stream = run({ files, coverage: true });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall());
+      stream.on('test:coverage', common.mustCall());
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and exclude by glob', async () => {
+      const stream = run({ files, coverage: true, coverageExcludeGlobs: ['test/*/test-runner/invalid-tap.js'] });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.some((path) => path.includes('test-runner/invalid-tap.js')), false);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
+
+    await it('should run with coverage and include by glob', async () => {
+      const stream = run({ files, coverage: true, coverageIncludeGlobs: ['test/*/test-runner/invalid-tap.js'] });
+      stream.on('test:fail', common.mustNotCall());
+      stream.on('test:pass', common.mustCall(1));
+      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
+        const filesPaths = files.map(({ path }) => path);
+        assert.strictEqual(filesPaths.some((path) => path.includes('test-runner/invalid-tap.js')), true);
+      }));
+      // eslint-disable-next-line no-unused-vars
+      for await (const _ of stream);
+    });
```

Comment:
```suggestion
    await it('should run with coverage and include by glob', async () => {
      const stream = run({ files: [
        ...files,
        'test/fixtures/test-runner/invalid-tap.js',
      ], coverage: true, coverageIncludeGlobs: ['test/*/test-runner/invalid-tap.js'] });
      stream.on('test:fail', common.mustNotCall());
      stream.on('test:pass', common.mustCall(1));
      stream.on('test:coverage', common.mustCall(({ summary: { files } }) => {
        const filesPaths = files.map(({ path }) => path);
        assert.deepStrictEqual(filesPaths, [fixtures.path('test-runner/invalid-tap.js')]);
      }));
      // eslint-disable-next-line no-unused-vars
      for await (const _ of stream);
    });
```

Reply:
Thats weird 🤔 
How was it able to ever pass then? 🤔

Reply:
Thanks!

Reply:
@RedYetiDev It still seems flaky.
Looking into it, but if you have additional input that would be appreciated 🙂

Reply:
@RedYetiDev Found the issue. the tests were running concurrently, which messed things up for some reason.